ProjectSidewalk/SidewalkWebpage
# Swap deprecated 'DOMNodeInserted' with Mutation Observer

Resolves #3616

I swapped the deprecated 'DOMNodeInserted' with a Mutation Observer instead, eliminating the error.

##### Before/After screenshots (if applicable)

**Before**
![image](https://github.com/user-attachments/assets/da9c86b4-77cc-45b2-8dd7-4a99ec6cd54d)
**After**
![image](https://github.com/user-attachments/assets/d962ee27-682a-49b6-a706-0e8b9228303b)

##### Testing instructions
1. Go to the gallery page with the dev tools console open, and see that the deprecation error is gone.
2. You can also add console.log statements before the event listeners are added to see that the mutation observer is properly firing.

##### Things to check before submitting the PR
